.github/deploy/setup_deploying_SPN.ps1

```diff
@@ -54,28 +54,40 @@ az ad app permission admin-consent --id $appId
 
 #####################################################################################
 Write-Host "Adding Owner rights. Needed to deploy resources." -ForegroundColor DarkGreen
-az role assignment create --assignee $appId --role "Owner" --subscription $account.id
+az role assignment create `
+  --assignee $appId `
+  --role "Owner" `
+  --subscription $account.id `
+  --scope "/subscriptions/$($account.id)"
 
 #######################################################################################
 Write-Host "Adding Microsoft graph permissions." -ForegroundColor DarkGreen
 # this is needed to be able to create other service principals for mounting
 
-$graph = az ad sp list --all | ConvertFrom-Json | Where-Object {$_.displayName -eq "Microsoft Graph"}
-
 # get the permission ID that we need:
-
 ## this was a detour that did not quite turn up the needed ID
-#$permissions = (az ad sp show --id $graph.appId | ConvertFrom-Json).oauth2permissions
+#$graph = az ad sp list --all --filter "displayName eq 'Microsoft Graph'" | ConvertFrom-Json
+#$permissions = (az ad sp show --id $graph.appId | ConvertFrom-Json).oauth2PermissionScopes
 #$permission = $permissions | Where-Object {$_.value -eq "Application.ReadWrite.All"}
```

Comment:
Is some of this outcommented code needed? or should it be removed?

Reply:
This code illustrates what we expected to be the right way to obtain the id of the permission that we are adding. I have also tried to run this code again and there is something missing. It does not work. But it is probably the right starting point to make it work. So I would like to keep it.